xblock/fields.py

```diff
@@ -965,29 +965,42 @@ def from_json(self, value):
             # Parser interprets empty string as now by default
             if value == "":
                 return None
+
             try:
                 value = dateutil.parser.parse(value)
             except (TypeError, ValueError):
                 raise ValueError("Could not parse {} as a date".format(value))
 
-        if not isinstance(value, datetime.datetime):
+        # Interpret raw numbers as a relative dates
+        if isinstance(value, (int, float)):
+            return datetime.timedelta(seconds=value)
```

Comment:
this line might be better written as:
`value = datetime.timedelta(seconds=value)`

That way the general flow is 1) interpret/transform `value`, 2) validate `value`'s type 3) fix up and return `value`.

Reply:
Oh, yeah, the `return` was a brainfart.